xrmocap/core/estimation/mview_sperson_smpl_estimator.py

```diff
@@ -401,10 +421,32 @@ def estimate_smpl(self, keypoints3d: Keypoints) -> SMPLData:
                 keypoints3d_conf=kps3d_conf,
                 keypoints3d_convention=working_convention,
                 handler_key='keypoints3d_limb_len'))
+
         registrant_output = self.smplify(
-            input_list=[kp3d_mse_input, kp3d_llen_input])
-        smpl_data = SMPLData()
+            input_list=[kp3d_mse_input, kp3d_llen_input],
+            init_param_dict=init_smpl_dict,
+            return_joints=return_joints,
+            return_verts=return_verts)
+
+        if self.smpl_data_type == 'smplx':
+            smpl_data = SMPLXData()
+        elif self.smpl_data_type == 'smpl':
+            smpl_data = SMPLData()
+        else:
+            self.logger.warning('smpl data type not defined, '
```

Comment:
Error (or warning) has already been told at L118, when building. We do not have to raise anything here.

Reply:
Fixed in [f33d0ca](https://github.com/openxrlab/xrmocap/pull/52/commits/f33d0caed97adb0a3c2afe553b5f9f34503f900f).